armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -427,6 +449,54 @@ def _getLinkedComponents(self, b, c):
             )
 
 
+def _determineLinked(componentA, componentB):
+    """determine axial component linkage for two components
+
+    Parameters
+    ----------
+    componentA : :py:class:`Component <armi.reactor.components.component.Component>`
+        component of interest
+    componentB : :py:class:`Component <armi.reactor.components.component.Component>`
+        component to compare and see if is linked to componentA
+
+    Notes
+    -----
+    - Requires that shapes have the getBoundingCircleInnerDiameter and getBoundingCircleOuterDiameter defined
+    - For axial linkage to be True, components MUST be solids, the same Component Class, multiplicity, and meet inner
+      and outer diameter requirements.
+
+    Returns
+    -------
+    linked : bool
+        status is componentA and componentB are axially linked to one another
+    """
+    if (
+        (componentA.containsSolidMaterial() and componentB.containsSolidMaterial())
+        and isinstance(componentA, type(componentB))
+        and (componentA.getDimension("mult") == componentB.getDimension("mult"))
+    ):
+        idA, odA = (
+            componentA.getBoundingCircleInnerDiameter(),
+            componentA.getBoundingCircleOuterDiameter(),
+        )
+        idB, odB = (
+            componentB.getBoundingCircleInnerDiameter(),
+            componentB.getBoundingCircleOuterDiameter(),
+        )
+
+        biggerID = max(idA, idB)
+        smallerOD = min(odA, odB)
+        if biggerID >= smallerOD:
```

Comment:
Elegant, nice! maybe ad a comment in the next line saying
#one object fits inside the other.

Reply:
Addressed. https://github.com/terrapower/armi/pull/670/commits/b5b468c5b7e1ea8838fbc9d7e933fa957c6701f9